test/Microsoft.Sbom.Parsers.Spdx22SbomParser.Tests/Parser/TestParser.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.Sbom.Contracts;
+using Microsoft.Sbom.Parsers.Spdx22SbomParser;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.Sbom.Parser
+{
+    internal class TestParser
+    {
+        private bool isFirstToken = true;
+        private JsonReaderState readerState;
+        private byte[] buffer = new byte[Constants.ReadBufferSize];
+
+        public IEnumerable<SBOMFile> GetFiles(Stream stream)
+        {        
+            stream.Read(buffer);
+
+            while (GetFiles(stream, out SBOMFile sbomFile) != 0)
+            {
+                yield return sbomFile;
+            }
+
+            long GetFiles(Stream stream, out SBOMFile sbomFile)
```

Comment:
A ref struct can't be used inside an iterator method, so used an inline method that has its own stack.